### PR TITLE
Add verbose option to echo that also prints the associated message info

### PR DESCRIPTION
**Pull request by ivanpauno**
Depends on https://github.com/ros2/rclpy/pull/922.

Exampler

``` bash
# terminal 1
ros2 run demo_nodes_py talker
```

```# terminal 2
ros2 topic echo /chatter -v
```

<details>
<summary>Expected output</summary>

```
message_info:
  source_timestamp: 1650489540621873147
  received_timestamp: 1650489540622072614
  publication_sequence_number: 857
message:
  data: 'Hello World: 856'
---
message_info:
  source_timestamp: 1650489541623013413
  received_timestamp: 1650489541623204404
  publication_sequence_number: 858
message:
  data: 'Hello World: 857'
---
message_info:
  source_timestamp: 1650489542623151805
  received_timestamp: 1650489542623384098
  publication_sequence_number: 859
message:
  data: 'Hello World: 858'
---
```
</details>

```
# terminal 2
ros2 topic echo /chatter -v --csv
```

<details>
<summary>Expected output</summary>

```
1650489559622070757,1650489559622579350,876,Hello World: 875
1650489560622630220,1650489560622846798,877,Hello World: 876
1650489561622649720,1650489561622884896,878,Hello World: 877
1650489562622548056,1650489562622715317,879,Hello World: 878
```
</details>

```# terminal 2
ros2 topic echo /chatter -v --raw
```

<details>
<summary>Expected output</summary>

```
---Got new message, message info:---     
{'source_timestamp': 1650489633622441430, 'received_timestamp': 1650489633622663794, 'publication_sequence_number': 950, 'reception_sequence_number': None}
---Message data:---
b'\x00\x01\x00\x00\x11\x00\x00\x00Hello World: 949\x00\x00\x00\x00'
---
---Got new message, message info:---
{'source_timestamp': 1650489634622511979, 'received_timestamp': 1650489634622743998, 'publication_sequence_number': 951, 'reception_sequence_number': None}
---Message data:---                      
b'\x00\x01\x00\x00\x11\x00\x00\x00Hello World: 950\x00\x00\x00\x00'
---     
---Got new message, message info:---
{'source_timestamp': 1650489635620004061, 'received_timestamp': 1650489635620098772, 'publication_sequence_number': 952, 'reception_sequence_number': None}
---Message data:---
b'\x00\x01\x00\x00\x11\x00\x00\x00Hello World: 951\x00\x00\x00\x00'
---
```
</details>

---

I'm not sure what to do in the case of the `--csv` option.
The yaml and raw outputs look reasonable to me.
The implementation is a bit tricky, I'm open to suggestions.

### CODE REVIEW

File: ros2topic/ros2topic/verb/echo.py [hunk 1]
```diff
@@ -122,6 +124,9 @@ def add_arguments(self, parser, cli_name):
                                                  'as well as m (the message).')
         parser.add_argument(
             '--once', action='store_true', help='Print the first message received and then exit.')
+        parser.add_argument(
+            '--verbose', '-v', action='store_true',
+            help='Shows the associated message info.')
```
[the remark this answers is not in the document]
Comment (ivanpauno) [reply ros2topic/ros2topic/verb/echo.py:129]: I don't mind, but the description can always be updated later.
I think it's good to mention what is exactly showing now when you use `--verbose/-v`.

**Review by wjwwood — APPROVED** [review]

**Review by fujitatomoya — APPROVED** [review]

**Review by jacobperron — APPROVED** [review]
LGTM

We could document that if you use `--csv` and `--verbose`, then it will prepend the following fields to the output: `source_timestamp,received_timestamp,publication_sequence_number`
And maybe document the timestamp units.

Hopefully that will make it a little bit clearer what to expect.

**Comment by wjwwood** [comment]
I already approved this, but a few thoughts:

- should the yaml use structures instead of nesting? https://yaml.org/spec/1.2.2/#22-structures

```yaml
---
source_timestamp: 1650489540621873147
received_timestamp: 1650489540622072614
publication_sequence_number: 857
---
data: 'Hello World: 856'
...
---
source_timestamp: 1650489541623013413
received_timestamp: 1650489541623204404
publication_sequence_number: 858
---
data: 'Hello World: 857'
...
```

- should the message info come over stderr instead and leave stdout for pure data? should that be configurable?
- should this get its own setting like `--include-message-info` rather than `--verbose`? it kind of seems like an abuse of verbose to me.

**Comment by ivanpauno** [comment]
> should this get its own setting like --include-message-info rather than --verbose? it kind of seems like an abuse of verbose to me.

Done in https://github.com/ros2/ros2cli/commit/657517dbe1664825e4aab83fbc7d6b9a42527527 as well.

> Should we always include reception_sequence_number/reception_sequence_number even if they're not supported (we could print 0 in that case, or even a "not supported" string)?

I modified that, now all the message info fields are printed, even if the rmw implementation doesn't support them.

**Comment by ivanpauno** [comment]
> We could document that if you use --csv and --verbose, then it will prepend the following fields to the output: source_timestamp,received_timestamp,publication_sequence_number

See https://github.com/ros2/ros2cli/pull/707/commits/785df9daec22201209b89d3e586916e6519ccfd4

**Comment by wjwwood** [comment]
> I don't think so, as it wouldn't be clear which message info corresponds to which message.
stdout/stderr buffers can be flushed at different times, the interleaving of message info and messages wouldn't be clear).

That's fair, I think it's good as is.

**Review by wjwwood — APPROVED** [review]

**Comment by wjwwood** [comment]
You could use the `...` in the YAML output to delineate the pairs of structures, but maybe that's not appropriate. I can't really tell from the reference.

It's not very important as it is relatively easy to isolate the pairs manually.

**Review by jacobperron — APPROVED** [review]
LGTM with green CI

**Comment by ivanpauno** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/16628/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/11210/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/17022/)

**Pull request merged**